post-processor/digitalocean-import/post-processor.go

```diff
@@ -0,0 +1,359 @@
+	Description  string        `mapstructure:"image_description"`
+	Distribution string        `mapstructure:"image_distribution"`
+	ImageRegions []string      `mapstructure:"image_regions"`
+	Timeout      time.Duration `mapstructure:"timeout"`
+
+	ctx interpolate.Context
+}
+
+type PostProcessor struct {
+	config Config
+}
+
+type apiTokenSource struct {
+	AccessToken string
+}
+
+type logger struct {
+	logger *log.Logger
+}
+
+func (t *apiTokenSource) Token() (*oauth2.Token, error) {
+	return &oauth2.Token{
+		AccessToken: t.AccessToken,
+	}, nil
+}
+
+func (l logger) Log(args ...interface{}) {
+	l.logger.Println(args...)
+}
+
+func (p *PostProcessor) Configure(raws ...interface{}) error {
+	err := config.Decode(&p.config, &config.DecodeOpts{
+		Interpolate:        true,
+		InterpolateContext: &p.config.ctx,
+		InterpolateFilter: &interpolate.RenderFilter{
+			Exclude: []string{"space_object_name"},
+		},
+	}, raws...)
+	if err != nil {
+		return err
+	}
+
+	if p.config.SpacesKey == "" {
+		p.config.SpacesKey = os.Getenv("SPACES_ACCESS_KEY")
+	}
+
+	if p.config.SpacesSecret == "" {
+		p.config.SpacesSecret = os.Getenv("SPACES_SECRET_KEY")
+	}
+
+	if p.config.APIToken == "" {
+		p.config.APIToken = os.Getenv("DIGITALOCEAN_API_TOKEN")
+	}
+
+	if p.config.ObjectName == "" {
+		p.config.ObjectName = "packer-import-{{timestamp}}"
+	}
+
+	if p.config.Distribution == "" {
+		p.config.Distribution = "Unkown"
+	}
+
+	if p.config.Timeout == 0 {
+		p.config.Timeout = 20 * time.Minute
+	}
+
+	errs := new(packer.MultiError)
+
+	if err = interpolate.Validate(p.config.ObjectName, &p.config.ctx); err != nil {
+		errs = packer.MultiErrorAppend(
+			errs, fmt.Errorf("Error parsing space_object_name template: %s", err))
+	}
+
+	templates := map[string]*string{
+		"api_token":     &p.config.APIToken,
+		"spaces_key":    &p.config.SpacesKey,
+		"spaces_secret": &p.config.SpacesSecret,
+		"spaces_region": &p.config.SpacesRegion,
+		"space_name":    &p.config.SpaceName,
+		"image_name":    &p.config.Name,
+		"image_regions": &p.config.ImageRegions[0],
+	}
+	for key, ptr := range templates {
+		if *ptr == "" {
+			errs = packer.MultiErrorAppend(
+				errs, fmt.Errorf("%s must be set", key))
+		}
+	}
+
+	if len(errs.Errors) > 0 {
+		return errs
+	}
+
+	packer.LogSecretFilter.Set(p.config.SpacesKey, p.config.SpacesSecret, p.config.APIToken)
+	log.Println(p.config)
+	return nil
+}
+
+func (p *PostProcessor) PostProcess(ui packer.Ui, artifact packer.Artifact) (packer.Artifact, bool, error) {
+	var err error
+
+	p.config.ObjectName, err = interpolate.Render(p.config.ObjectName, &p.config.ctx)
+	if err != nil {
+		return nil, false, fmt.Errorf("Error rendering space_object_name template: %s", err)
+	}
+	log.Printf("Rendered space_object_name as %s", p.config.ObjectName)
+
+	log.Println("Looking for image in artifact")
+	source := ""
+	validSuffix := []string{"raw", "img", "qcow2", "vhdx", "vdi", "vmdk", "bz2", "tar.xz", "tar.gz"}
```

Comment:
`s/bz2/tar.bz2/` ?

Reply:
Strangely, the API will reject files suffixed with `.tar.bz2` as an unsupported format but not ones w/ just `.bz2` That is likely a bug on the DigitalOcean side, but it does beg the question of whether this check is completely necessary. Failing fast is generally a good idea, but maybe we should just let the API enforce this? 

Are there builders likely to output additional files? The other purpose of the loop is to make sure that if there is more than one file in `artifact.Files()`, we find the correct one.

Reply:
I'm slightly leaning towards skipping this check and just link to the api docs for importing images. But I let @SwampDragons decide.

Reply:
I agree with Rickard -- Let's skip this check. Failing fast is great, but this won't protect users in the validation stage of the Packer run. They'll already have gone through an entire build before failing, which means this doesn't buy us much.

Reply:
What I'm curious about is if it is safe for me to just take `artifact.Files()[0]` It's not clear to me if it is common for builders to output multiple files. I suppose I could keep the check, but if nothing is found with one of those extensions, log a warning and then just take `artifact.Files()[0]`

Reply:
OK. Pushed another commit (https://github.com/hashicorp/packer/pull/7060/commits/ee430d9a40753e4e814bb9f915b07ca73b13c462). Now we first check how many files `artifact.Files()` returns, and if there is only one, we use it. If there are multiple files, then we use the filter to make sure we upload the correct one. This still means there is a list of extensions to maintain here, but it guards against blindly uploading something with no idea if it is an image file.

Sound good?

Reply:
Sure. I'm fine with that.